src/java.base/share/classes/java/lang/Integer.java

```diff
@@ -453,16 +454,102 @@ private static void formatUnsignedIntUTF16(int val, int shift, byte[] buf, int l
      */
     @IntrinsicCandidate
     public static String toString(int i) {
-        int size = stringSize(i);
-        if (COMPACT_STRINGS) {
-            byte[] buf = new byte[size];
-            getChars(i, size, buf);
-            return new String(buf, LATIN1);
-        } else {
+        if (!COMPACT_STRINGS) {
+            int size = stringSize(i);
             byte[] buf = new byte[size * 2];
             StringUTF16.getChars(i, size, buf);
             return new String(buf, UTF16);
         }
+
+        if (i == Integer.MIN_VALUE) {
```

Comment:
The more common practice is to negate positive numbers. In this way, we can remove this branch and maintain the same style as the other methods in this file.